static/src/javascripts/projects/common/modules/discussion/loader.js

```diff
@@ -0,0 +1,589 @@
+                thisLoader.setCommentHash(id);
+            }
+        } else if (!fromRequest) {
+            // If the comment isn't on the page, then we need to load the comment thread
+            thisLoader.loadComments({ comment: id });
+        } else {
+            // The comment didn't exist in the response
+
+            // Scroll to toolbar and show message
+            scrollToElement(qwery('.js-discussion-toolbar'), 100);
+
+            fastdom.write(() => {
+                $('.js-discussion-main-comments').prepend(
+                    '<div class="d-discussion__message d-discussion__message--error">The comment you requested could not be found.</div>'
+                );
+            });
+
+            // Capture in sentry
+            raven.captureMessage("Comment doesn't exist in response", {
+                level: 'error',
+                extra: {
+                    commentId: id,
+                },
+            });
+        }
+    }
+
+    gotoPage(page: number): void {
+        scrollToElement(qwery('.js-discussion-toolbar'), 100);
+        this.comments.relativeDates();
+        this.loadComments({ page });
+    }
+
+    loadComments(
+        options?: {
+            comment?: string | number,
+            shouldTruncate?: boolean,
+            page?: number,
+        } = {}
+    ): Promise<void> {
+        const opts = Object.assign({}, options);
+
+        this.setState('loading');
+
+        // If the caller specified truncation, do not load all comments.
+        if (
+            opts &&
+            opts.shouldTruncate &&
+            this.comments.isAllPageSizeActive()
+        ) {
+            opts.pageSize = 10;
+        }
+
+        // Closed state of comments is passed so we bust cache of comment thread when it is reopened
+        opts.commentsClosed = this.getDiscussionClosed();
+
+        return this.comments.fetchComments(opts).then(() => {
+            this.removeState('loading');
+
+            if (opts && opts.shouldTruncate) {
+                this.setState('truncated');
+            } else {
+                // do not call removeTruncation because it could invoke another fetch.
+                this.removeState('truncated');
+            }
+
+            if (this.comments.shouldShowPageSizeMessage()) {
+                this.setState('pagesize-msg-show');
+            } else {
+                this.removeState('pagesize-msg-show');
+            }
+
+            if (opts.comment) {
+                this.gotoComment(opts.comment.toString(), true);
+            }
+        });
+    }
+
+    removeTruncation(): void {
+        // When the pagesize is 'All', the full page is not yet loaded, so load the comments.
+        if (this.comments.isAllPageSizeActive()) {
+            this.loadComments();
+        } else {
+            this.removeState('truncated');
+        }
+    }
+
+    renderCommentBox(elem: HTMLElement): void {
+        return new CommentBox({
```

Comment:
I don't think we need a return statement here, as the return value will always be `undefined`